src/Worker/PushSubscription.php

```diff
@@ -58,31 +60,29 @@ public static function execute(int $sid, int $nid)
 			return;
 		}
 
-		$user = User::getById($notification['uid']);
+		$user = User::getById($Notification->uid);
 		if (empty($user)) {
 			Logger::info('User not found', ['application' => $subscription['uid']]);
 			return;
 		}
 
 		$l10n = DI::l10n()->withLang($user['language']);
 
-		$type = Notification::getType($notification);
-
-		if (!empty($notification['actor-id'])) {
-			$actor = Contact::getById($notification['actor-id']);
+		if ($Notification->actorId) {
+			$actor = Contact::getById($Notification->actorId);
 		}
 
 		$body = '';
 
-		if (!empty($notification['target-uri-id'])) {
-			$post = Post::selectFirst([], ['uri-id' => $notification['target-uri-id'], 'uid' => [0, $notification['uid']]]);
+		if ($Notification->targetUriId) {
+			$post = Post::selectFirst([], ['uri-id' => $Notification->targetUriId, 'uid' => [0, $Notification->uid]]);
 			if (!empty($post['body'])) {
 				$body = BBCode::toPlaintext($post['body'], false);
-				$body = Plaintext::shorten($body, 160, $notification['uid']);
+				$body = Plaintext::shorten($body, 160, $Notification->uid);
 			}
 		}
 
-		$message = Notification::getMessage($notification);
+		$message = (new Notifications\Factory\Notification(DI::logger()))->getMessageFromNotification($Notification, DI::baseUrl(), $l10n);
```

Comment:
I just scrolled through the code and saw this line - what is this? Why is it so complicate?

Reply:
@MrPetovan - add the whole class to the `src/DI.php` as a new method and add `baseUrl` as a constructor argument, so the call would be:
```php
$message = DI::notificationFactory()->getMessageFromNotification($Notification, l10n);
```
So Dice will become responsible for creating the right dependencies :-)
At least we now would see, how a "manual" dependency injection would work :-)

Reply:
Another point: Dice caches shared objects :-)